### PR TITLE
add package: cmark 0.27.1

### PR DESCRIPTION
**Pull request by mklein994**
This is really useful for viewing READMEs:

``` console
$ cmark -t html README.md > README.html
$ python -m http.server
$ # open localhost:8000/README.html in browser
```

### CODE REVIEW

File: packages/cmark/build.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		-DCMAKE_AR=`which ${TERMUX_HOST_PLATFORM}-ar` \
+		-DCMAKE_BUILD_TYPE=MinSizeRel \
+		-DCMAKE_C_FLAGS="$CFLAGS $CPPFLAGS" \
+		-DCMAKE_CROSSCOMPILING=True \
+		-DCMAKE_CXX_FLAGS="$CXXFLAGS" \
+		-DCMAKE_FIND_ROOT_PATH=$TERMUX_PREFIX \
+		-DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY \
+		-DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY \
+		-DCMAKE_INSTALL_PREFIX=$TERMUX_PREFIX \
+		-DCMAKE_LINKER=`which ${TERMUX_HOST_PLATFORM}-ld` \
```
Comment (vishalbiswas) [inline packages/cmark/build.sh]: The linker is set in env var `LD`

File: packages/cmark/build.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		-DCMAKE_MAKE_PROGRAM=`which make` \
+		-DCMAKE_RANLIB=`which ${TERMUX_HOST_PLATFORM}-ranlib` \
```
Comment (vishalbiswas) [inline packages/cmark/build.sh]: `RANLIB` is also exported

**Comment by mklein994** [comment]
Honestly, I built this package from within Termux setting pretty much only the prefix, and everything worked fine, so I just modified the `build.sh` script from weechat. I'm not too familiar with cmake and its nuances at the moment.

**Comment by TheDiamondYT1** [comment]
@vishalbiswas @fornwall Is there a file or wiki page which lists and explains all the possible entries in build.sh?

**Comment by vishalbiswas** [comment]
@TheDiamondYT1 Unfortunately, no. But I highly recommend reading and understanding the `build-package.sh` file, and so does the wiki.

**Comment by TheDiamondYT1** [comment]
Am i supposed to build a package from termux or from linux? It doesnt work from termux.

**Comment by its-pointless** [comment]
from linux. You can build all the packages in reasonable amount of time. Only clang takes a long time ....

**Comment by mklein994** [comment]
This last change fixes the build (tested through docker), but note that was **forced pushed** to rebase with upstream.

**Comment by fornwall** [comment]
Thanks a lot! The package has now been built and is available for installation:

    apt update && apt install cmark

**Pull request merged**